dockerhub-prune-tags.py

```diff
@@ -68,36 +68,38 @@ def set_filter(fn, seq):
     return set(filter(fn, seq))
 
 def authstr(user, passwd):
-    from base64 import encodestring
-    return encodestring('%s:%s' % (user,passwd)).replace('\n', '')
+    from base64 import b64encode, encode
+    auth = f'{user}:{passwd}'
+    auth_bytes = p.encode()
+    return str(b64encode(auth_bytes))
```

Comment:
Hi @mtru32,

A few things here:

- that is cool that `base64.b64encode` does not append a trailing newline
- it doesn't look like `encode` is used anywhere
- it doesn't look like `p` is _defined_ anywhere (won't even compile as a result)
- converting a `bytes` object with `str(...)` will give the `b'...'`, which is not what we want - you've gotta decode it to `str`

I think we can fix these with

```suggestion
    from base64 import b64encode
    auth = f'{user}:{passwd}'
    auth_bytes = auth.encode()
    return b64encode(auth_bytes).decode()
```

Reply:
Hey @edquist thanks for the comments. My mistake on the 'p' variable, that was the result of renaming too quickly. Let me know if you need anything else!